sample-apps/liberty-app/src/test/java/org/example/app/LibertySmokeIT.java

```diff
@@ -0,0 +1,32 @@
+package org.example.app;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.microshed.testing.MicroShedApplication;
+import org.microshed.testing.health.Health;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Requires an local environment which is already running at {@code localhost:9080}.
+ * Thus enables to decouple the life cycles of the tests and results in a smaller turnaround time.
+ * <p>
+ * Note that this is a different approach to {@link LibertyAppSmokeIT}.
+ */
+public class LibertySmokeIT {
+
+    private MicroShedApplication application;
+
+    @BeforeEach
+    void setUp() {
+        application = MicroShedApplication.withBaseUri("http://localhost:9080").build();
```

Comment:
If the application is already running at `localhost:9080` that's fine, but we shouldn't hard-code that into the test source. Instead, the host and port should be set with `microshed_hostname` and `microshed_http_port` system/env properties and then many sources (such as ApplicationContainer and RestClientBuilder) can auto-detect those values for "local dev" mode, but also switch to running with Testcontainers instances for when this runs in TravisCI.

That last part (also running in CI) is important IMO. If I understand this code correctly, these tests would fail in CI because they require something else to start the server?

Reply:
Yes, wasn't sure how to fix that, please change so it doesn't interfere with the projects pipeline. In a real-world project I'd also rather make it aware of e.g. System properties or so :)